lua/lazyvim/plugins/extras/lang/java.lua

```diff
@@ -182,6 +192,11 @@ return {
         callback = function(args)
           local client = vim.lsp.get_client_by_id(args.data.client_id)
           if client and client.name == "jdtls" then
+            -- INFO: without this the inlay hints start toggled off by default.
+            -- Probably because `nvim-lspconfig` is not used to set up `nvim-jdtls`
+            -- and `LazyVim.toggle.inlay_hints` doesn't execute, so we have
+            -- to execute it again ourselves in this autocmd in `LspAttach`
+            LazyVim.toggle.inlay_hints(args.buf, true)
```

Comment:
You can probably remove these lines now and inlay hints would still work if you have it enabled in lsp-config.
Would be great if you can test this.

Make sure you're using the main version of LAzyVim.  See tip at https://www.lazyvim.org/configuration/lazy.nvim

Reply:
I just tested it locally with crappy GPT java code and inlay hints are enabled by default (not from the PR but with personal configuration in `nvim-jdtls` spec like I had mentioned in the Discussion thread, so it should be the same). Will take a look at the new functionalities you added tomorrow. I'm in awe of your work as always :smile:

Reply:
I just refactored the code to be more vim like as well. Great that it works now!!

Reply:
Yes, but don't take my word for granted and wait for @idelice's input as he's the one who deals with Java and he (and maybe other users) should test the new behavior.

Reply:
@folke I opened up two different java projects, and I could confirm that the inlay hints are being triggered as soon as the jdtls have been attached! 

<img width="506" alt="Screenshot 2024-05-20 at 10 15 45" src="https://github.com/LazyVim/LazyVim/assets/59118341/9a66625e-933b-4789-adfd-c242c1cf4180">

<img width="908" alt="Screenshot 2024-05-20 at 10 15 59" src="https://github.com/LazyVim/LazyVim/assets/59118341/448b51db-2e3c-42f2-b0ff-a880ee1c2929">

But for some reason, in some places, the inlay hints are not working. I don't know if it's related to lazyvim or the api itself but here's a screenshot of where the inlay hints should be triggered (inside the `new User(....)`)

<img width="388" alt="Screenshot 2024-05-20 at 10 16 34" src="https://github.com/LazyVim/LazyVim/assets/59118341/b352a1b5-a2cd-4190-be3e-36c3edf0bc03">

Reply:
Did they get triggered with this PR and LazyVim release?

Reply:
I used the latest development release as @folke asked me to, and it worked fine with my local jdtls changes:
 
```lua
      settings = {
        java = {
          inlayHints = { parameterNames = { enabled = "all" } },
```

I also tried to use local forked LazyVim in my lazy config:

```lua
return { "LazyVim/LazyVim", dev = true, dir = "path/to/local/lazyvim" }
```

But it returned this when opening nvim:
<img width="607" alt="Screenshot 2024-05-20 at 11 47 22" src="https://github.com/LazyVim/LazyVim/assets/59118341/4c8b9458-ecfb-4061-afe5-7aa0161904dc">

Reply:
To use local forked repo I just have this in my `/lua/config/lazy.lua`
```lua
  dev = {
    path = "~/projects/plugins",
    patterns = { "LazyVim" },
  },
```
You can change the `path` to whatever local path you prefer.
I also have created another LazyVim profile under directory `~/.config/LazyVimDev` and call that with `NVIM_APPNAME=LazyVimDev nvim` when I want to use the local forked repo, so that it doesn't mess with my main LazyVim profile.

Reply:
Ok I finally managed to find my way around the docs to run lazyvim locally... anyway, it seems that I'm now getting more inlay hints than before but with the cost of losing the inlay hints are the very first file it opens - which means, if i close and reopen the same buffer - the inlay hints will appear.

Reply:
<img width="520" alt="Screenshot 2024-05-20 at 16 34 44" src="https://github.com/LazyVim/LazyVim/assets/59118341/78233fe2-bf9c-431f-afa0-7ea6df6c0569">

Reply:
@folke - your latest development release I tried before - actually triggered the inlay hints when jdtls first attached.

Reply:
**Development branch**

After testing with the development branch, it seems that the inlay hints are not appearing in every place anymore. Please see attachment and compare with the attached image from my previous comment:

<img width="391" alt="Screenshot 2024-05-20 at 16 58 19" src="https://github.com/LazyVim/LazyVim/assets/59118341/301219ea-4695-439c-ac45-f0e4eac68ff7">

There's clearly a big difference.

Reply:
Did you make any changes to make them appear as in the first image?

Reply:
No. The _only_ difference is the branches i'm using.

The first one from `main` and the other `release-please--branches--main`.

Reply:
The `main` branch is the one you should be testing against. It contains the changes that have been merged in the latest HEAD and will be incorporated in the next release. So, with the `main` branch everything works as it should?

Reply:
Yes

Reply:
Do you still have to close and reopen the first buffer? Can you instead do `:e`? Or maybe `:LspRestart`? One of the 2 I imagine should do the trick instead of closing and reopening the buffer.

Reply:
> Do you still have to close and reopen the first buffer? Can you instead do `:e`? Or maybe `:LspRestart`? One of the 2 I imagine should do the trick instead of closing and reopening the buffer.

Yes unfortunately. I haven't tried :e or anything else. I can do that later and report back

Reply:
Did you try that with the suggestion of Folke by removing the suggested lines? Do you observe any different behavior if you don't remove the suggested lines? Whenever you have free time.

Reply:
Removing that line does no difference at all from what i could see. Even in the main branch.

I still had to reopen the buffer or `:e` before inlay hints would appear.

The most noticable difference between the main and the dev branch is that the main branch somehow shows _more_ inlay hints - which is GOOD! 

But i still have to :e or reopen the buffer in order for inlay hints to appear or just open another java file.